tasks/gotest.py

```diff
@@ -943,3 +945,46 @@ def lint_go(
     only_modified_packages=False,
 ):
     raise Exit("This task is deprecated, please use `inv linter.go`", 1)
+
+
+@task
+def check_otel_dependencies(ctx):
+    with ctx.cd("test/otel"):
+        # Rename fixtures
+        ctx.run("mv dependencies.go.fake dependencies.go")
+        ctx.run("mv go.mod.fake go.mod")
+        ctx.run("mv go.sum.fake go.sum")
+
+        # Update dependencies to latest local version
+        ctx.run("go mod tidy")
+
+        # Build test/otel/dependencies.go with same settings as `make otelcontribcol`
+        res = ctx.run("GO111MODULE=on CGO_ENABLED=0 go build -trimpath -o . .", warn=True)
+        if res is None or not res.ok:
+            ctx.run("mv dependencies.go dependencies.go.fake")
+            ctx.run("mv go.mod go.mod.fake")
+            ctx.run("mv go.sum go.sum.fake")
+            raise Exit(f"Error building otel components with datadog-agent dependencies: {res.stderr}")
+
+        # Remove built executable and rename fixtures
+        ctx.run("rm ./otel")
+        ctx.run("mv dependencies.go dependencies.go.fake")
+        ctx.run("mv go.mod go.mod.fake")
+        ctx.run("mv go.sum go.sum.fake")
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You could use python os API instead, also above. You could also copy the file and let it there, the runner will clean them up automatically, and if you add the generated file yo `.gitignore` you don't care about them locally

Reply:
Thanks for the suggestions, I decided to use shutil instead because it has a copy method